codeChunks.Rmd

```diff
@@ -0,0 +1,156 @@
+```{r eval=F}
+i=0
+allRows <- c() # initialize vector 
+for(i in 1:nrow(dataFrame)){
+  allRows[i] <- all(is.na(dataFrame[i,])) # store each output into allRows
+}
+
+blankRows <- which(allRows) # outputs indices of rows that contain "TRUE" (rows with all NA's) 
+dataFrame <- dataFrame[-blankRows,] # remove those blankRows from dataFrame
+```
+
+Alternatively, you can use apply() to iterate through all rows. You can use this for a single data frame or a list of multiple data frames using a for loop. 
+```{r eval=F}
+# outputs indices of rows with all NA's
+blankRows <- which(apply(dataFrame,1,function(x)all(is.na(x)))) 
+```
+
+##Columns
+```{r eval=F}
+
+i=0
+allCols <- c()
+for(i in 1:length(dataFrame)){ # length(dataFrame) gives us # of cols
+  allRows[i] <- all(is.na(dataFrame[,i])) # notice that we switch where the i goes 
+}
+
+blankCols <- which(allCols)
+dataFrame <- dataFrame[,-blankCols]
+```
+
+Alternatively:
+```{r eval=F}
+blankCols <- which(apply(dataFrame,2,function(x)all(is.na(x))))
+```
+
+
+#Searching Through Strings - Dates
+
+Use the grepl() function to search for a particular string. Since we often have to reformat dates in our data sets, searching for particular dates or times could be useful. 
+
+```{r}
+# an example of common date/time scenarios
+# this is usually a column within a data frame
+dates <- c("3/4/2016", "3/4/16", "3-4-2016", "3-4-16","3-4-16 12:30",
+           "3/4/2016", "3/4/16", "3-4-2016", "3-4-16","3-4-16 12:30",
+           "3/4/2016", "3/4/16", "3-4-2016", "3-4-16","3-4-16 12:30")
+```
+
+Run unique() to see what kind of formats there are. 
+```{r}
+unique(dates)
+```
+The international standard format for dates and time are YYYY-MM-DD and hh:mm:ss respectively, while the combined date-time standard is YYYY-MM-DDThh:mm:ss. Often times, researchers' data contain dates and times in varying formats because it may have been inputted by different people. 
+
+None of these are in the standard format, so we'll have to do some reformatting. 
+
+
+The following code gives us the indices that contain "/2016".
+```{r}
+indDates <- which(grepl("/2016",dates))
+indDates
+```
+
+Use as.POSIXct() to specify what our original date format is. 
+Use format() to specify the format that we want.
+Store values back into dates vector. 
+```{r}
+dates[indDates] <- format(as.POSIXct(dates[indDates], tz = "", format="%m/%d/%Y"), format = "%Y-%m-%d")
+dates[indDates]
+```
+This same process works for all of the formats in our dates vector.
+Note: 
+"-16" is ambiguous because it could also refer to the day within an already standard-formatted date (e.g. 2018-05-16). Always check to make sure. 
+We will reformat combined date/time items before the observations that don't contain times because they also contain "-16", which is ambiguous.
+
+```{r}
+indDates1 <- which(grepl("/16",dates))
+dates[indDates1] <- format(as.POSIXct(dates[indDates1], tz = "", format="%m/%d/%y"), format = "%Y-%m-%d")
```

Comment:
I like to use the `lubridate` package to work with dates. If you haven't tried it, I'd definitely recommend checking it out!  https://cran.r-project.org/web/packages/lubridate/vignettes/lubridate.html There are also some other date/time packages, but I'm not as familiar with them. `tibbletime` is another one seems promising.